### PR TITLE
feat: typescript 3.8, nullish coalescing, optional chaining

### PR DESCRIPTION
**Pull request by jacekkarczmarczyk**
## Motivation and Context
Ability to use new ts features

### bash
```bash
# Failing tests fail also with 3.5
yarn test directives
yarn test effects
yarn test framework
yarn test locale
yarn test services
yarn test util
```
### playground
```vue
<script lang="ts">
  import { h, defineComponent } from 'vue'

  export default defineComponent({
    setup (props) {
      const foo = { bar: 1 }

      return () => h('div', [
        foo?.bar?.baz,
        foo.noMethod?.(),
        foo.fizz ?? 'ok',
      ])
    },
  })
</script>

```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,7 +52,7 @@
     "shelljs": "^0.8.3",
     "ts-jest": "^24.0.2",
     "tslint": "^5.18.0",
-    "typescript": "^3.5.3",
+    "typescript": "^3.8.0-dev.20200201",
```
Comment (KaelWD) [inline package.json]: https://github.com/microsoft/TypeScript/releases/tag/v3.8-rc

File: packages/vuetify/package.json [hunk 2]
```diff
@@ -123,5 +125,8 @@
     "tags": "dist/json/tags.json",
     "attributes": "dist/json/attributes.json"
   },
-  "web-types": "dist/json/web-types.json"
+  "web-types": "dist/json/web-types.json",
+  "dependencies": {
+    "typescript": "^3.8.0-dev.20200206"
```
Comment (KaelWD) [inline packages/vuetify/package.json]: why dis here

File: packages/vuetify/babel.config.js [hunk 3]
```diff
@@ -54,4 +54,6 @@ if (['lib', 'es5'].includes(env)) {
 
 if (env !== 'lib') {
   module.exports.plugins.push('@babel/plugin-proposal-object-rest-spread')
+  module.exports.plugins.push('@babel/plugin-proposal-optional-chaining')
```
Comment (KaelWD) [inline packages/vuetify/babel.config.js]: Use loose mode? https://babeljs.io/docs/en/babel-plugin-proposal-optional-chaining#loose

File: package.json (hunk 1 above)
Comment (jacekkarczmarczyk) [reply package.json]: I'll wait for 3.8.0 (should be yesterday)

File: packages/vuetify/babel.config.js [hunk 4]
```diff
@@ -53,5 +53,17 @@ if (['lib', 'es5'].includes(env)) {
 }
 
 if (env !== 'lib') {
-  module.exports.plugins.push('@babel/plugin-proposal-object-rest-spread')
+  module.exports.plugins.push([
+    '@babel/plugin-proposal-object-rest-spread',
+    {
+      loose: true,
```
Comment (KaelWD) [inline packages/vuetify/babel.config.js]: Wrong line?

File: packages/docs/babel.config.js [hunk 5]
```diff
@@ -5,5 +5,19 @@ module.exports = {
       modules: false,
       corejs: { version: 2, proposals: true }
     }]
+  ],
+  plugins: [
+    [
+      '@babel/plugin-proposal-optional-chaining',
+      {
```
Comment (KaelWD) [inline packages/docs/babel.config.js]: \*autistic screeching\*
```js
{ loose: true }
```

**Pull request merged**